### ISSUE TITLE
ResourceNotFoundException when using collection

### ISSUE THREAD

**Issue by angeliand** [paulcwarren/spring-content#186]
Hi!
First of, thank you for this library, it has saved me a lot of time. :) 
My situation:
I want to store multiple images for a product with Spring Content FS. Uploading the images works fine and when I upload only 1 image, downloading it is also working. However when a product has multiple images associated with it, I get Resource Not Found. 
Image:
```
@Embeddable
@Getter
@Setter
@NoArgsConstructor
public class Image {

    @JsonView(Views.Public.class)
    @ContentId
    private UUID contentId;

    @ContentLength
    @JsonView(Views.Public.class)
    private Long contentLength;

    @MimeType
    @JsonView(Views.Public.class)
    private String mimeType;
}
```
Product:
```
@Data
@Entity
@Table(name = "PRODUCTS")
@EntityListeners(AuditingEntityListener.class)
public class Product implements Serializable {

    @Id
    @Column(name = "PRODUCT_ID")
    @GeneratedValue
    @JsonView(Views.Public.class)
    private int id;

    @RestResource
    @ElementCollection(fetch = FetchType.EAGER)
    @JsonView(Views.Public.class)
    private List<Image> images = new ArrayList<>();
     // other attributes....
}
```
ImageContentStore:
```
@Component
@StoreRestResource(path = "products")
public interface ImageContentStore extends ContentStore<Image, UUID>{
}
```
ProductRepository:
```
@RepositoryRestResource(path = "products")
public interface ProductRepository extends JpaRepository<Product, Integer> {
}
```
I have set the baseUri to /content. 
When downloading the image (React frontend) I use the {apiUrl}/content/products/{productId}/images/{contentId} url-s, and as I have said I get the image if I uploaded only one. 

I am using spring 2.2, and spring content 1.0.0.M9.

**Comment by angeliand** [comment]
Also (I do not know if these are related) I had to make my own Converter, because when downloading the image I got an error which said there exists no converter for Image -> String.
```
@Bean
    public FileSystemResourceLoader fsResourceLoader() {
        return new FileSystemResourceLoader(filesystemRoot().toString());
    }

    public Converter<Image, String> converter() {
        return new FilesystemStoreConverter<Image, String>() {

            @Override
            public String convert(Image source) {
                if(source.getContentId()==null){
                    source.setContentId(UUID.randomUUID());
                }
                return source.getContentId().toString();
            }
        };
    }

    public Converter<UUID, String> uuidConverter() {
        return new FilesystemStoreConverter<UUID, String>() {

            @Override
            public String convert(UUID uuid) {
                return uuid.toString();
            }
        };
    }

    @Bean
    public FilesystemStoreConfigurer configurer() {
        return new FilesystemStoreConfigurer() {

            @Override
            public void configureFilesystemStoreConverters(ConverterRegistry registry) {
                registry.addConverter(converter());
                registry.addConverter(uuidConverter());
            }
        };
    }
```

**Comment by paulcwarren** [comment]
Hi @angeliand, thanks for giving spring content a go.  What you describe sounds like a bug or a misconfiguration.  Let me try and reproduce and we can go from there.

**Comment by paulcwarren** [comment]
Hi @angeliand,

When I first read this I hadn't spotted that you are using @Embedded/@Embeddable entities.  So the links we are discussing here are "property" content links; i.e. a link that allows you to access the content of a child entity through its parent entity (because, in this case, the entity that the content is actually associated with isn't URL addressable itself).  

At one point in time, spring-content did support these so-called "property" content links and vestiges of that may remain in some examples but we are trying to clean up.  

However, after a time, we thought the use case was actually a bit of an edge case after all.  Do people use @Embedded for real?  Plus confusing as it isn't supported uniformly across all databases.  JPA requires annotations to express this type of relationship but other databases do not which makes the implementation somewhat complex.  Therefore, we stepped back from that and today the entity has to be addressable; i.e. have a Repository and be exported.  This fits better with Spring Data REST too.

So, I've updated your example and [commited here](https://github.com/paulcwarren/issue-186) to show you it working ok when Product and Image are modeled as Entities, each with their own Repositories.

Based on that though I would like to ask if you are you using @Embedded for real?  Or were you, perhaps, following some examples we left out there in the wild and, in fact, modeling your domain model as Entities with Repositories is ok for your use case?

**Comment by angeliand** [comment]
Hi!

I modified my code to include your suggestions, tried it out and it works! I cannot thank you enough. 

I was not using @Embedded originally, I tried to make it work following the example in [Property Collection Associations](https://paulcwarren.github.io/spring-content/refs/snapshot/fs-index.html#_property_collection_associations), but the urls listed did not work for me (although I did not have a Repository for images, I assumed I did not need one).  I started digging and I found issue #69 and rewrote everything to use Embedded according to the linked tests. 

But thanks to you it's working nicely now :)
